packages/next-tinacms-github/src/get-github-static-props.ts

```diff
@@ -0,0 +1,70 @@
+/**
+
+Copyright 2019 Forestry.io Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+import { GithubError } from './github/content/GithubError'
+import { getMarkdownFile } from './github/content/getMarkdownFile'
+import { getJsonFile } from './github/content/getJsonFile'
+
+export interface PreviewMeta {
+  github_access_token: string
+  fork_full_name: string
+  head_branch: string
+  file: {
+    relativePath: string
+    name?: string
+    format?: 'markdown' | 'json'
+  }
+}
+
+export async function getGithubStaticProps(
+  preview: boolean,
```

Comment:
Do we need to pass preview through?

Reply:
Struggled on that one. I went with "yes" for consistencies sake. That is, consistency with `getStaticProps`. You could basically do this:

```ts
export const getStaticProps = getGithubStaticProps
```

Reply:
As is, that would fail to load content unless in edit-mode. preview prop-wise, not a big deal though

Reply:
Hmm, I just noticed the APIs didn't match anyway. 

What if we rename it to `getGithubPreviewProps` and then remove the `preview` flag from the first arg:

export const getStaticProps: GetStaticProps = async function({
  preview,
  previewData,
}) {
  if (preview) {
    return getGithubPreviewProps({
      ...previewData,
      fileRelativePath: 'src/content/home.json',
      format: 'json'
    });
  }
  return {
    props: {
      sourceProvider: null,
      previewError: null,
      preview,
      file: {
        fileRelativePath: 'src/content/home.json',
        data: (await import('../content/home.json')).default,
      },
    },
  };
}

Reply:
Yeah makes sense to me

Reply:
Done